cmd/kube-controller-manager/app/core.go

```diff
@@ -201,6 +208,11 @@ func newNodeLifecycleControllerDescriptor() *ControllerDescriptor {
 }
 
 func startNodeLifecycleController(ctx context.Context, controllerContext ControllerContext, controllerName string) (controller.Interface, bool, error) {
+	if controllerContext.Cloud == nil {
+		klog.FromContext(ctx).Info("Warning: node-controller is set, but no cloud provider specified. Will not configure node lifecyle controller.")
+		return nil, false, nil
+	}
+
```

Comment:
this does not need cloud and this is the one breaking, you confused this with `newCloudNodeLifecycleControllerDescriptor`
 You can identify the controllers that need cloud not nil via the initFunc in the constructor that has `isCloudProviderController`

```

func newServiceLBControllerDescriptor() *ControllerDescriptor {
	return &ControllerDescriptor{
		name:                      cpnames.ServiceLBController,
		aliases:                   []string{"service"},
		initFunc:                  startServiceLBController,
		isCloudProviderController: true,
	}
}
```

Reply:
the one you have to change is just below
```

func newCloudNodeLifecycleControllerDescriptor() *ControllerDescriptor {
	return &ControllerDescriptor{
		name:                      cpnames.CloudNodeLifecycleController,
		aliases:                   []string{"cloud-node-lifecycle"},
		initFunc:                  startCloudNodeLifecycleController,
		isCloudProviderController: true,
	}
}
func startCloudNodeLifecycleController(ctx context.Context, controllerContext ControllerContext, controllerName string) (controller.Interface, bool, error) {
	logger := klog.FromContext(ctx)
```